### ISSUE TITLE
Long tweets fail with error Tweet needs to be a bit shorter (Code 186)

### ISSUE THREAD

**Issue by tcm390** [elizaOS/eliza#1178]
When attempting to send tweets longer than 280 characters using the Eliza Twitter client, the API responds with an error:

```
Error sending tweet; Bad response: {
  errors: [
    {
      message: 'Authorization: Tweet needs to be a bit shorter. (186)',
      locations: [Array],
      path: [Array],
      extensions: [Object],
      code: 186,
      kind: 'Permissions',
      name: 'AuthorizationError',
      source: 'Client',
      tracing: [Object]
    }
  ],
  data: {}
} 
```

https://discord.com/channels/1253563208833433701/1300025221834739744/1318559898312904745

<img width="551" alt="Screenshot 2024-12-17 at 8 20 34 AM" src="https://github.com/user-attachments/assets/83a25ae7-355c-4ba3-ad8a-04c6c0dd04d3" />

**Comment by tcm390** [comment]
> Hi [@tcm390](https://github.com/tcm390) could you add a direct link to the message for all issues gotten from discord. This is to help with investigation since there might have been some conversation around it.

yes, updated.

**Comment by oxSaturn** [comment]
This would be the same issue as https://github.com/ai16z/eliza/issues/883.

**Comment by konstantine25b** [comment]
okay but is it fixed? there was written merged but I dont think it works yet. Can please tell me if it works or not ? Thanks inadvance

**Comment by tcm390** [comment]
> okay but is it fixed? there was written merged but I dont think it works yet. Can please tell me if it works or not ? Thanks inadvance

It hasn’t been released yet. We need to wait until the latest scraper is available. I’ve already asked the team to release it, but they haven’t responded yet.

**Comment by shakkernerd** [comment]
agent-twitter-client v0.0.17 has been released. You may have to pull changes from `develop` branch or update the version of agent-twitter-client manually on your local version.

**Comment by konstantine25b** [comment]
okay thank you very much for informing me about new update and fixing that issue.

**Comment by konstantine25b** [comment]
@shakkernerd so i pulled changes but there is no function called sendLongTweet - which was my issue. I mean there is sendTweet and there should be sendLongTweet for tweets more than 280 character long size but it is not there can you check it please

**Comment by shakkernerd** [comment]
Implemented in #1339
